web_src/js/markup/tasklist.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Attaches `change` handlers to markdown rendered tasklist checkboxes in comments.
+ * When a checkbox value changes, the corresponding [ ] or [x] in the markdown string is set accordingly and sent to the server.
+ * On success it updates the raw-content on error it resets the checkbox to its original value.
+ */
+export default function initMarkupTasklist() {
+  $(`.render-content.markup[data-can-edit='true']`).parent().each((_, container) => {
+    const $container = $(container);
+    const $checkboxes = $container.find(`.task-list-item input:checkbox`);
+
+    $checkboxes.on('change', async (ev) => {
+      const $checkbox = $(ev.target);
+      const checkboxCharacter = $checkbox.is(':checked') ? 'x' : ' ';
+      const position = parseInt($checkbox.data('source-position')) + 1;
+
+      const $rawContent = $container.find('.raw-content');
+      const oldContent = $rawContent.text();
+      const newContent = oldContent.substring(0, position) + checkboxCharacter + oldContent.substring(position + 1);
+
+      if (newContent !== oldContent) {
+        $checkboxes.prop('disabled', true);
+
+        try {
+          const $contentZone = $container.find('.edit-content-zone');
+          const url = $contentZone.data('update-url');
+          const context = $contentZone.data('context');
+
+          await $.post(url, {
+            _csrf: window.config.csrf,
+            content: newContent,
+            context,
+          });
+
+          $rawContent.text(newContent);
+        } catch (e) {
+          $checkbox.prop('checked', !$checkbox.is(':checked'));
+
+          console.error(e);
+        } finally {
+          $checkboxes.prop('disabled', false);
+        }
+      }
+    });
+
+    $checkboxes.prop('disabled', false);
+  });
+}
```

Comment:
It's mostly just a preference from me but I'd like new features to be written with vanilla JS instead of jQuery because I'd like to eventually get rid of the jQuery dependency. Modern Browser APIs have made jQuery obsolete and can in some cases perform up to 20 times faster than jQuery.

Reply:
I agree with you. I replaced the code with vanilla JS but kept the `$.post` because there is no "inline FormData" expression like the object.
What's your opinion about the class vs. data topic?

Reply:
`$.post` is fine, I want to refactor all those cases to use `fetch` eventually.

HTML class would be useful for CSS in this case as we can add a hover effect to the checkbox border if it's ediable, giving the user a hint they can interact with it. Regular checkboxes already have a similar effect, we should replicate it for in-markdown checkboxes too.

Reply:
Feel free to edit the PR.

Reply:
You may need to add me as collaborator on your fork.

Reply:
You can resolve this thread.